packages/@atjson/document/src/annotation.ts

```diff
@@ -275,26 +276,17 @@ export abstract class Annotation<Attributes = {}> {
       // with the preserveAdjacentBoundaries switch.
 
       // Default edge behaviour.
-    } else if (change.behaviour === AdjacentBoundaryBehaviour.default) {
-      if (change.start === this.start) {
+    } else if (change.start === this.start) {
+      if (change.behaviourLeading === EdgeBehaviour.preserve) {
         this.start += length;
         this.end += length;
-      } else if (change.start === this.end) {
+      } else {
```

Comment:
Feel like that'll make this part of the function a little confusing / clunky since the trailing edge one would have no behavior for preserve but happy to do that if it adds more clarity

Reply:
Nah, it's all fine